src/main/java/org/truffleruby/core/module/ModuleFields.java

```diff
@@ -79,9 +79,16 @@ public static void debugModuleChain(DynamicObject module) {
     private boolean hasFullName = false;
     private String name = null;
 
+    private boolean isRefinement = false;
+    private boolean isOverlaid = false;
+    private DynamicObject refinedClass;
+    private DynamicObject definedAt;
+
     private final ConcurrentMap<String, InternalMethod> methods = new ConcurrentHashMap<>();
     private final ConcurrentMap<String, RubyConstant> constants = new ConcurrentHashMap<>();
     private final ConcurrentMap<String, Object> classVariables = new ConcurrentHashMap<>();
+    private final ConcurrentMap<DynamicObject, DynamicObject> refinements = new ConcurrentHashMap<>();
+    private final ConcurrentMap<DynamicObject, DynamicObject> activatedRefinements = new ConcurrentHashMap<>();
```

Comment:
What's activated refinements? Could you add a comment about it?
Also what is they key and the value, same for `refinements`.

Reply:
Commented added and more explanation: https://github.com/graalvm/truffleruby/pull/860#discussion_r157344825